postgres/cloudpostgres/cloudpostgres.go

```diff
@@ -55,6 +58,45 @@ type Params struct {
 	TraceOpts []ocsql.TraceOption
 }
 
+// URLOpener opens GCP PostgreSQL URLs
+// like "cloudpostgres://user:password@myproject/us-central1/instanceId/mydb".
+type URLOpener struct {
+	// TraceOpts contains options for OpenCensus.
+	TraceOpts  []ocsql.TraceOption
+	CertSource *certs.RemoteCertSource
+}
+
+// Scheme is the URL scheme cloudpostgres registers its URLOpener under on
+// postgres.DefaultMux.
+const Scheme = "cloudpostgres"
+
+func init() {
+	postgres.DefaultURLMux().RegisterPostgres(Scheme, &URLOpener{})
+}
+
+// OpenPostgresURL opens a new GCP database connection wrapped with OpenCensus instrumentation.
+func (uo *URLOpener) OpenPostgresURL(ctx context.Context, u *url.URL) (*sql.DB, error) {
+	vals := u.Query()
+	u.RawQuery = vals.Encode()
+	gcpInstance := strings.Split(u.Path, "/")
```

Comment:
I believe you meant:

```suggestion
	gcpInstance := strings.Split(u.Host + u.Path, "/")
```

I think it would be worth it to split out the string-splitting code into its own function so we can unit test that part, which would catch these sort of trip-ups.

Reply:
Actually there is no need to do so, coz `u.Host` will take `ProjectId` part by default, the only reason why we can do like this it's just to avoid misunderstanding.